From 097cfe01588570f924050bc3d1423a0266cd1ca4 Mon Sep 17 00:00:00 2001
From: Laine Stump <laine@laine.org>
Date: Fri, 7 Feb 2014 16:17:24 +0200
Subject: [PATCH] Require bridge-utils for netcf-libs in the specfile

This resolves:

  https://bugzilla.redhat.com/show_bug.cgi?id=1060317 (RHEL7)
  https://bugzilla.redhat.com/show_bug.cgi?id=911403 (Fedora 20)

netcf itself doesn't use anything in bridge-utils, but it called
/sbin/ifup, which *can* run brctl (which is part of bridge-utils).
Technically, initscripts should require brutils (since /sbin/ifup is a
part of initscripts), but we all know that's not going to
happen. Instead we can just make netcf require bridge-utils. It's not
a very large package, so it's unlikely we'll get any complaints about
bloat.

(NB: in the past, libvirt had a Requires: bridge-utils, so this didn't
used to be a problem. libvirt removed this Requires: in v0.9.8, when
it switched from execing brctl to directly calling ioctls to create
and manage bridge devices.)

(cherry picked from commit 5bb22ec48096338a6651590299740f9e42367bb9)
---
 netcf.spec.in | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/netcf.spec.in b/netcf.spec.in
index 6ab3ee5..7862e74 100644
--- a/netcf.spec.in
+++ b/netcf.spec.in
@@ -66,6 +66,10 @@ developing applications that use %{name}.
 Summary:        Libraries for %{name}
 Group:          System Environment/Libraries
 
+# bridge-utils is needed because /sbin/ifup calls brctl
+# if you create a bridge device
+Requires:       bridge-utils
+
 %description    libs
 The libraries for %{name}.
 
-- 
1.8.3.1

